test/widgets/cards/greeting_card_test.dart

```diff
@@ -0,0 +1,110 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_screenutil/flutter_screenutil.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:network_image_mock/network_image_mock.dart';
+import 'package:tackleapp/widgets/cards/greeting_card.dart';
+
+void main() {
+  group('GreetingCard Widget Tests', () {
+    const testGreetingMessage = 'Hello, Welcome!';
+    const testUserName = 'John Doe';
+    const testProfilePictureUrl = 'https://example.com/profile-picture.jpg';
+
+    Widget createWidgetUnderTest({
+      required String greetingMessage,
+      required String userName,
+      required String profilePictureUrl,
+    }) =>
+        ScreenUtilInit(
+          designSize: const Size(440, 956),
+          minTextAdapt: true,
+          builder: (context, child) => MaterialApp(
+            home: Scaffold(
+              body: GreetingCard(
+                greetingMessage: greetingMessage,
+                userName: userName,
+                profilePictureUrl: profilePictureUrl,
+              ),
+            ),
+          ),
+        );
+
+    testWidgets(
+        'Given valid greeting message, user name, and profile picture URL '
+        'when GreetingCard is rendered '
+        'Then it should display the greetingmessage, username, profilepicture',
+        (WidgetTester tester) async {
+      // Act
+      await mockNetworkImagesFor(
+        () async {
+          await tester.pumpWidget(
+            createWidgetUnderTest(
+              greetingMessage: testGreetingMessage,
+              userName: testUserName,
+              profilePictureUrl: testProfilePictureUrl,
+            ),
+          );
+        },
+      );
+
+      // Assert
+      expect(find.text(testGreetingMessage), findsOneWidget);
+      expect(find.text(testUserName), findsOneWidget);
+      expect(find.byType(CircleAvatar), findsOneWidget);
+    });
+
+    testWidgets(
```

Comment:
remove this test case as only first name should be visible.
Maybe you can try giving a very long first name

Reply:
done